cancer_ai/validator/competition_handlers/melanoma_handler.py

```diff
@@ -0,0 +1,56 @@
+from .base_handler import BaseCompetitionHandler
+from .base_handler import ModelEvaluationResult
+
+from PIL import Image
+import numpy as np
+from sklearn.metrics import accuracy_score, precision_score, recall_score, confusion_matrix, roc_curve, auc
+
+class MelanomaCompetitionHandler(BaseCompetitionHandler):
+    """
+    """
+    def __init__(self, path_X_test, y_test) -> None:
+        super().__init__(path_X_test, y_test)
+
+    def preprocess_data(self):
+        X_test = []
+        target_size=(224, 224) #TODO: Change this to the correct size 
+
+        for img_path in self.path_X_test:
+            img = Image.open(img_path)
+            img = img.resize(target_size)
+            img_array = np.array(img, dtype=np.float32) / 255.0
+            img_array = np.array(img)  
+            if img_array.shape[-1] != 3:    # Handle grayscale images
+                img_array = np.stack((img_array,) * 3, axis=-1)
+            
+            img_array = np.transpose(img_array, (2, 0, 1))           # Transpose image to (C, H, W)
+
+            X_test.append(img_array)
+
+        X_test = np.array(X_test, dtype=np.float32)
+
+        # Map y_test to 0, 1
+        y_test = [1 if y == "True" else 0 for y in self.y_test]
+
+        return X_test, y_test
+    
+    def evaluate(self, y_test, y_pred, run_time) -> ModelEvaluationResult:
```

Comment:
może bardziej "analyze_model_result" czy coś takiego, evaluate oznacza że to to odpalasz, a tutaj nie odpalasz

Reply:
get_model_result może, nie wiem wymyśl coś ;)

Reply:
Ready